### DISCUSSION TITLE
data_string

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#18] [Announcements]
# 字符串和文件读写

https://lv.qizhen.xyz/data_string

**Comment by Arita9527** [comment]
为啥字符串---数字转科学计数法函数，使用小数出来的结果，整数部分大于1啊？不是应该小于1吗？
![](https://pic1.zhimg.com/80/v2-f2c682b681fe8ac982073a576df950f8_720w.jpg)

**Reply by ruanqizhen** [reply]
没太看明白你的问题，你是问，为什么科学计数法表示有个整数部分吗？科学计数法通常是有一位整数位（1~9），加上小数位，加上幂次。或者你是问工程计数法？

**Reply by Arita9527** [reply]
看来是我搞混了。
原来科学计数法和工程计数法是不一样的东西。

科学记数法，是将数字表示成10的幂的倍数的形式。
而工程记数法，是在科学记数法基础上，将10的幂限制为**3的倍数**，并且倍数在1到1000之间（不含1000）

我还一直纳闷说，为什么123.40 E-12里面的整数部分是123，原来工程计数法E后面一定要是3的倍数。感谢大神

**Comment by zsl-nuaa** [comment]
请问节后的练习示例代码有吗

**Reply by zsl-nuaa** [reply]
用“一维数组排序” 中文姓名和英文姓名是分开的单独排序怎么解决 例如结果为bob、tom、阿、李、张   我想要的结果是“阿”排第一 然后是bob.......

**Reply by ruanqizhen** [reply]
我还没有为练习题编写示例代码。通常，你运行一下自己编写的程序，如果结果正确就可以了。

关于名字排序，LabVIEW 本身并不知道字符转中是中文还是英文，它只是按照字符的编码顺序来排序。中文字符的编码在数值上都大于英文字母的编码，因此会排在后面。

你提出的要求是按照中文拼音排序。所以，首先要得到每一个中文字符的拼音。LabVIEW 没有任何处理中文的能力，但是市面上有不少开源免费的模块可以做这件事。你可以找一个免费的 python 库，在 LabVIEW 中调用相应 python 函数，把中文转换成拼音。然后再按照拼音排序即可。

**Reply by zsl-nuaa** [reply]
好的 感谢
